### ISSUE TITLE
Document "Move to group" menu

### ISSUE THREAD

**Issue by CaseyKazan** [Quicksaver/Tab-Groups#164]
1) I understand that Tab Groups is easier to use than multiple browser windows.  However, I use an extended desktop on two screens (laptop and external).  This allows me to keep two browser windows fully visible without having to swap back and forth (just a quick mouse swipe)--it's faster than clicking on the Tab Groups icon and selecting and loading another group (even on one monitor, switching open browser windows is actually faster than switching groups in one window).  However, Tab Groups only has tabs for that browser window but doesn't share/show them in any other browser window (and vice-versa).  This would be a great feature add to see all Tab Groups in all open browser windows.

2) Also, it would be great if Tab Groups would really function with Firefox Sync.  I know the work-around of bookmarking an entire group, naming it, and then being able to open up the whole bookmark group but that's really cumbersome and involves another step--creating a new tab group.

3) Which brings me to my next idea: How about when clicking the Tab Groups icon, that the window includes the user's existing groups (if any) plus a pre-set group that opens up either with a blank tab/home tab or pre-populated with tabs (like Google Calendar, favorite search engine, news, whatever) that could be set in the Tab Groups Settings menu.  This would allow two things: one--to create a new group; and, two--to open up a bookmarks folder without having to disturb the contents of an existing group.

Thanks.

**Comment by Quicksaver** [comment]
1) #82.
2) #44.
3) ... what? I really don't understand the point. Can't you just create a group with all those tabs and select it whenever you want? Or better yet, use pinned tabs. Am I missing something?

**Comment by CaseyKazan** [comment]
Hey, Luis:
    
    1) Issue #82 does not address sharing groups between browser
    windows, only the ability to drag a tab from one window to another. 
    My suggestion is that ALL of the groups "sync" when you open up a
    new window, just like all of the Firefox settings function the same
    and bookmarks appear in every Firefox window one opens.  (Which also
    raises the idea of having Tab Groups be able to have sub-groups. 
    E.g. I create a group called "Shopping" with tabs for Ebay, Amazon,
    etc.  But I can also create a sub-group in "Shopping" called "Shoes"
    with tabs for Zappos, etc.  This way, the user can have LOTS of
    groups and sub-groups without ending up with microscopic icons in
    the groups view or having big bunches of unrelated tabs opening
    (E.g. if shoe shopping, not have one's car shopping, electronics
    shopping, etc. also opening up).  Think of it like Tab Groups
    working in parallel with Firefox Bookmarks but only with the user's
    desired groups.)
    
    2) Issue #44:  yes, exactly.  Firefox Sync would fully synchronize
    groups and tabs across devices, not just Tab Groups settings.  As it
    is, Firefox (possibly because of Tab Groups?) doesn't actually sync
    tabs at all across devices even if that option is turned on in
    Firefox options.
    
    3) Let me try again:  never mind!  I just re-discovered the
    additional Tab Groups option when right-clicking on a tab: Move to
    Group > (list of group names) / New Group (option on bottom of
    drop-down menu).  [Now that I know this one thing, I like Tab Groups
    17% more.]  Okay, maybe add this information to the Tab
    Groups Options > How To Use page.  I didn't even know that this
    was added on to the tab right-click menu!  This won't even take any
    programming, just some text editing!  :-)
    
    4) Finally, if I don't go back to the forum on GitHub, how will I
    know if this (or anything else) is added to Tab Groups?  Would you
    create a new version that would be installed via Firefox >
    Add-ons > Settings > Update Add-ons Automatically?
    
    I know, so many words, so many questions....
    
    Evan
    
    
    
    On 3/25/2016 1:47 PM, Luís Miguel
      wrote:
    
    
      1) #82.
        2) #44.
        3) ... what? I really don't understand the point. Can't you just
        create a group with all those tabs and select it whenever you
        want? Or better yet, use pinned tabs. Am I missing something?
      —
        You are receiving this because you authored the thread.
        Reply to this email directly or view
          it on GitHub

**Comment by iShyboy** [comment]
For sync. https://github.com/ReDEnergy/SessionSync this man made a awesome addon that works with firefox bookmarks, if you could save a group as a folder in the addon you could restore the groups from the folder of the addon.
The addon alone, see the session as a whole, but just in the active window, it doesn't see the groups.
![sessionsync](http://i.imgur.com/o0xwOWh.png)

Tab groups would just save the groups as bookmarked folders in the addon main folder.
And you could just in another window, restore that folder.
This would be a immediate solution, but with a bit of cooperation between the devs, we could see a feature from session sync to restore selected folders as a group (in background of course).

**Comment by Quicksaver** [comment]
> My suggestion is that ALL of the groups "sync" when you open up a new window, just like all of the Firefox settings function the same and bookmarks appear in every Firefox window one opens.

Unfortunately that's not possible. Tabs themselves aren't shared like that, making a change like that goes a little beyond the scope of this add-on. Even implementing #82 will be a challenge as it is.

> Which also raises the idea of having Tab Groups be able to have sub-groups. 

You're probably better off just using completely separate groups for all those examples, or even using multiple windows with multiple windows in that case, although I understand those are not exactly the same things you were looking for.

> maybe add this information to the Tab Groups Options > How To Use page.  I didn't even know that this was added on to the tab right-click menu!  This won't even take any programming, just some text editing!  :-)

True. :) I'll keep this issue open to remind me of this point.

> how will I know if this (or anything else) is added to Tab Groups?

Yes, that should always apply the latest version, and every time the add-on updates it should show you a changelog, which is in the About pane of the add-on's preferences (unless you disable that option of course).

**Comment by Quicksaver** [comment]
@iShyboy I admit I don't fully follow your suggestion, but that's probably because I don't know or haven't tried that add-on yet. I'll take a look at it once I start working on #82, but from your description I don't think any of that will be necessary.

**Comment by iShyboy** [comment]
No harm, bookmark one group as a bookmark folder, solve the problem between windows, I already can do this, with just more steps, but it's not like I use more than one window, I used to the single window mode of TMP, I use this for a few years already.
